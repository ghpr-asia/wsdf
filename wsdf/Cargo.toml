[package]
name = "wsdf"
version = "0.1.0"
edition = "2021"
autotests = false
license = "MIT OR Apache-2.0"
description = "Wireshark Dissector Framework"
homepage = "https://github.com/ghpr-asia/wsdf"
documentation = "https://docs.rs/wsdf"
repository = "https://github.com/ghpr-asia/wsdf"
readme = "README.md"
keywords = ["wireshark", "dissectors"]
categories = ["network-programming"]

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dev-dependencies]
trybuild = { version = "1.0", features = ["diff"] }

[dependencies]
wsdf-derive = { path = "../wsdf-derive", version = "0.1.0" }
epan-sys = { path = "../epan-sys", version = "0.1.0" }
thiserror = "1.0"

[[example]]
name = "udp"
crate-type = ["cdylib"]

[[example]]
name = "moldudp64"
crate-type = ["cdylib"]

[[example]]
name = "dns"
crate-type = ["cdylib"]
