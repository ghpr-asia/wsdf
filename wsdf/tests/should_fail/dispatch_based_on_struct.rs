// Tests that if we try to dispatch on a struct value that it fails. There is no  *real* reason for
// this apart from keeping our implementation simple for now.

use wsdf::*;

#[derive(Protocol)]
#[wsdf(decode_from = "moldudp.payload")]
struct ProtoFoo {
    bar: Bar,
    #[wsdf(dispatch_field = "bar")]
    qux: Qux,
}

struct Bar {
    baz: u32,
}

enum Qux {
    Hot,
    Cold,
}

impl Qux {
    fn dispatch_bar(bar: &Bar) -> usize {
        0
    }
}

fn main() { }
