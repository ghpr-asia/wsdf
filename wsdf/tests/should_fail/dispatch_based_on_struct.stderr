error: this field cannot be used to dispatch enums
 --> tests/should_fail/dispatch_based_on_struct.rs:9:5
  |
9 |     bar: Bar,
  |     ^^^
